libp2p/wire.nim

```diff
@@ -20,21 +20,45 @@ else:
 
 const
   TRANSPMA* = mapOr(
-    mapAnd(IP, mapEq("udp")),
-    mapAnd(IP, mapEq("tcp")),
-    mapAnd(mapEq("unix"))
+    UDP,
+    TCP,
+    UNIX,
   )
 
   RTRANSPMA* = mapOr(
-    mapAnd(IP, mapEq("tcp")),
-    mapAnd(mapEq("unix"))
+    TCP,
+    UNIX
   )
 
+proc resolveDns(ma: MultiAddress, domain: Domain = Domain.AF_UNSPEC, prefix: string = ""): MaResult[TransportAddress]
+  {.raises: [Defect, ResultError[string]]} =
+  try:
+    var
+      dnsbuf: array[256, byte]
+      dnsval: string
+      pbuf: array[2, byte]
+    if ma[0].tryGet().protoArgument(dnsbuf).tryGet() == 0:
```

Comment:
Probably better to use `ma[0].isOk...`

Reply:
since this function returns `Result`, `(? ma[0]).protoArgument...`